server/api/users/users.controller.ts

```diff
@@ -29,9 +29,32 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 };
 
 export const getUsers: RequestHandler<UserProfile[]> = async (req, res) => {
```

Comment:
i just realized this line - what is this `UserProfile[]` trying to type?  i believe the first argument for `RequestHandler` is the params.  if you're trying to type the request response, you'll have to also include the pagination info